src/viam/app/data_client.py

```diff
@@ -60,6 +62,54 @@ class DataClient:
     `ViamClient`.
     """
 
+    class TabularData:
+        """Class representing a piece of tabular data and associated metadata.
+
+        Args:
+            data (Mapping[str, Any]): the requested data.
+            metadata (viam.proto.app.data.CaptureMetadata): the metadata from the request.
+            time_requested (datetime): the time the data request was sent.
+            time_received (datetime): the time the requested data was received.
+        """
+
+        def __init__(self, data: Mapping[str, Any], metadata: CaptureMetadata, time_requested: datetime, time_received: datetime) -> None:
+            self.data = data
+            self.metadata = metadata
+            self.time_requested = time_requested
+            self.time_received = time_received
+
+        data: Mapping[str, Any]
+        metadata: CaptureMetadata
+        time_requested: datetime
+        time_received: datetime
+
+        def __str__(self) -> str:
+            return f"{self.data}\n{self.metadata}Time requested: {self.time_requested}\nTime received: {self.time_received}\n"
+
+        def __eq__(self, other: "DataClient.TabularData") -> bool:
+            return str(self) == str(other)
+
```

Comment:
Thanks for commenting on that! I don't think implementing the custom `str` or `eq` methods for the others is necessary at this time. I added the custom `str` here because we want to print out meaningful information when saving the data to file, and I added the `eq` method because we need it for testing. Because those concerns aren't relevant for the other shadow classes at this time, I don't think there's any need to do worry about adding these methods.

Re: adding a `proto` property, I'd love to do so but these aren't really shadow types in quite the same way. That is, there's no 1:1 mapping from the `BinaryData` or `TabularData` types that I implemented to a single proto-defined type (there are parameters we still don't care about from the proto type, the proto response actually contains two lists of different types and we're grabbing pieces of data from each, etc.). So I don't think a `proto` property really makes sense in this case.